tests/test_main.py

```diff
@@ -42,6 +42,54 @@ def test_read_text(tmp_path):
     assert sql_query == "SELECT 1 AS patient_id"
 
 
+@pytest.mark.parametrize(
+    "sql_query,are_t100s_handled",
+    [
+        # not handled by query
+        (
+            """
+                SELECT Patient_ID FROM Patient
+            """,
+            False,
+        ),
+        # handled by query
+        (
+            f"""
+                SELECT Patient_ID FROM Patient
+                WHERE Patient_ID NOT IN (SELECT Patient_ID FROM {T100S_TABLE})
+            """,
+            True,
+        ),
+        # not handled by comment
+        (
+            """
+                -- t100s intentionally not excluded
+                SELECT Patient_ID FROM Patient
+            """,
+            False,
+        ),
+        # handled by comment
+        (
+            f"""
+                -- {T100S_TABLE} intentionally not excluded
+                SELECT Patient_ID FROM Patient
+            """,
+            True,
+        ),
+        # handled by query, but undesirable: it's hard to prevent this with a "light
+        # touch" approach
+        (
+            f"""
+                SELECT Patient_ID FROM {T100S_TABLE}
+             """,
+            True,
+        ),
```

Comment:
This is undesirable when it's the only query, but desirable when it's a common table expression. Whilst we could change `are_t100s_handled` to distinguish between undesirable and desirable, I don't think substring matching is up to the job: we'd need something more powerful (e.g. [SQLGlot](https://sqlglot.com/sqlglot.html)), which would be much less "light touch".

Reply:
I agree; if we're trying to prevent well-meaning users accidentally forgetting to handle t1oos, we have to accept that this would be possible.